packages/markdown/loader.js

```diff
@@ -178,9 +206,14 @@ module.exports = async function demoLoader() {
         raw: await fs.readFile(moduleFilepath, { encoding: 'utf8' }),
       };
       demoModuleIDs.add(moduleID);
-      extractImports(demos[demoName].raw).forEach((importModuleID) =>
-        importedModuleIDs.add(importModuleID),
-      );
+      extractImports(demos[demoName].raw).forEach((importModuleID) => {
+        // detect relative import
+        // skip for modules that are not demos
+        if (!toolbarHidden.has(moduleID.replace('./', ''))) {
```

Comment:
So I guess this could ship as its own PR optimizing things?

Reply:
Do you mean taking this part apart into a separate PR focussed only on optimisation? I'd imagine this can stay, whereas future optimisation can go in a new PR

Reply:
I think we should have this as a standalone win/PR. If you go to https://mui.com/material-ui/react-grid/#interactive. Why are we loading the source to make that demo editable? Doesn't make sense.